rest/src/main/java/io/syndesis/rest/v1/util/Urls.java

```diff
@@ -17,17 +17,33 @@
 
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Optional;
 
 import javax.servlet.http.HttpServletRequest;
 
 public final class Urls {
 
+    /* default */ static final String ORIGIN_HEADER = "X-Forwarded-Origin";
+
     private Urls() {
         // Utility class, only static methods
     }
 
+    public static URI absoluteTo(final HttpServletRequest httpRequest, final URI url) {
```

Comment:
Does it make more sense to use URL rather than URI? We're always going to be working with URLs here right, rather than URIs?